Adyen/Utilities/External/LocalizableRegionRepository.swift

```diff
@@ -0,0 +1,359 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// Model for any geographic region.
+public struct Region: Decodable, CustomStringConvertible, Equatable {
+
+    /// Unique identifier.
+    public let identifier: String
+
+    /// Localised human-friendly name.
+    public let name: String
+
+    /// :nodoc:
+    public var description: String {
+        name
+    }
+
+    /// :nodoc:
+    internal enum CodingKeys: String, CodingKey {
+        case identifier = "id"
+        case name
+    }
+}
+
+// swiftlint:disable type_body_length
+
+/// Fetch localized geographic regions from external.
+public class RegionRepository {
+
+    private let environment: Environment
+
+    private var dataTask: URLSessionDataTask?
+
+    private static var cache = NSCache<NSURL, NSArray>()
+
+    /// Create new instance of RegionRepository.
+    /// - Parameters:
+    ///   - environment: The environment to use for base URL.
+    public init(environment: Environment) {
+        self.environment = environment
+    }
+
+    /// Request list of countries, translated to a selected language.
+    /// - Parameters:
+    ///   - locale: The locale for translation.
+    ///   - callback: The callback to receive countries.
+    public func getCountries(locale: String, callback: @escaping (([Region]) -> Void)) {
+        let url = self.url(for: ["countries", locale])
+        if let cachedValue = RegionRepository.cache.object(forKey: url as NSURL) as? [Region] {
+            callback(cachedValue)
```

Comment:
you should return here, right? no need to loadResource again

Reply:
👍